src/main/java/org/datadog/jenkins/plugins/datadog/util/git/GitUtils.java

```diff
@@ -111,6 +115,39 @@ public static GitMetadata buildGitMetadata(@Nullable final GitClient gitClient)
         }
     }
 
+    /**
+     * When checking out a repository for a multi-branch pipeline,
+     * Jenkins does so in "detached HEAD" state.
+     * This means that a specific commit is checked out, rather than a branch.
+     * To work around this the plugin examines available branches
+     * to see which one of them points to the checked out commit.
+     * <p>
+     * It is possible that multiple branches point to the same commit.
+     * In which case the logic in this method tries to use a "hint" extracted from environment variables.
+     */
+    private static String chooseBranch(@Nonnull Collection<String> branches, @Nullable String branchHint, String buildName) {
```

Comment:
Any possibility of adding unit tests for this method?

Reply:
Moved this logic to the repo callback